data-avro/src/main/java/com/linkedin/data/avro/DataTranslator.java

```diff
@@ -609,18 +609,17 @@ private Object translate(Object value, DataSchema dataSchema, Schema avroSchema)
             Schema fieldAvroSchema = avroField.schema();
             Object fieldValue = map.get(fieldName);
             boolean isOptional = field.getOptional();
-            if (isOptional  || (
-                _dataTranslationOptions != null &&
-                ((DataMapToAvroRecordTranslationOptions) _dataTranslationOptions). getDefaultFieldDataTranslationMode()
-                    == PegasusToAvroDefaultFieldTranslationMode.DO_NOT_TRANSLATE))
+            if (isOptional)
```

Comment:
I originally left this comment, which I'm still curious about.

> Based on my understanding, this if-statement is just used to determine if the field schema was converted to a union. This doesn't seem to be consistent with the logic that actually determines whether to convert the schema field to a union.

Reply:
After arguing for a few hours, I've come to the conclusion that the way you're doing it here is valid.